src/Microsoft.Framework.PackageManager/Restore/NuGet/NuGetv2Feed.cs

```diff
@@ -146,25 +146,35 @@ public Task<IEnumerable<PackageInfo>> FindPackagesByIdAsync(string id)
                 }
                 catch (Exception ex)
                 {
-                    if (retry == 2)
+                    var isFinalAttempt = (retry == 2);
+                    var message = ex.Message;
+                    if (ex is TaskCanceledException)
+                    {
+                        message = ErrorMessageUtils.GetFriendlyTimeoutErrorMessage(
+                            ex as TaskCanceledException,
+                            isFinalAttempt,
+                            ignoreFailure: false);
+                    }
+
+                    if (isFinalAttempt)
```

Comment:
I don't think this should be around the _ignoreFailure check since we don't want to do those 3 times